From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon Sep 5 15:01:00 2016 +0200
Subject: [PATCH] bcm2835: Add Raspberry Pi thermal sensor to the device tree
Patch-mainline: Not yet, posted on linux-rpi-kernel
References: bsc#989511

Driven via the Raspberry Pi VideoCore 4 firmware interface.

Signed-off-by: Lubomir Rintel <lkundrak@v3.sk>
Signed-off-by: Fabian Vogt <fvogt@suse.com>
Cc: Stephen Warren <swarren@wwwdotorg.org>
Cc: Lee Jones <lee@kernel.org>
Cc: Eric Anholt <eric@anholt.net>
Cc: devicetree@vger.kernel.org
Cc: linux-rpi-kernel@lists.infradead.org
Cc: linux-arm-kernel@lists.infradead.org
---
Needs the RPi firmware patchset from branch 'rpi-firmware' of 
https://github.com/anholt/linux

 Documentation/devicetree/bindings/thermal/raspberrypi,bcm2835-thermal.txt |   13 ++++++++++
 arch/arm/boot/dts/bcm2835-rpi.dtsi                                        |    5 +++
 2 files changed, 18 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/thermal/raspberrypi,bcm2835-thermal.txt

--- /dev/null
+++ b/Documentation/devicetree/bindings/thermal/raspberrypi,bcm2835-thermal.txt
@@ -0,0 +1,13 @@
+Raspberry Pi Broadcom BCM2835 thermal control
+
+Required properties:
+
+- compatible : should be "raspberrypi,bcm2835-thermal"
+- firmware : the Raspberry Pi firmware node
+
+Example:
+
+thermal {
+	compatible = "raspberrypi,bcm2835-thermal";
+	firmware = <&firmware>;
+};
--- a/arch/arm/boot/dts/bcm2835-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2835-rpi.dtsi
@@ -26,6 +26,11 @@
 			firmware = <&firmware>;
 			#power-domain-cells = <1>;
 		};
+
+		thermal {
+			compatible = "raspberrypi,bcm2835-thermal";
+			firmware = <&firmware>;
+		};
 	};
 };
 
